backends/dpdk/main.cpp

```diff
@@ -117,6 +117,25 @@ int main(int argc, char *const argv[]) {
         p4rt->serializeBFRuntimeSchema(out);
     }
 
+    if (!options.tdiFile.isNullOrEmpty()) {
+        auto p4RuntimeSerializer = P4::P4RuntimeSerializer::get();
+        if (options.arch == "psa")
```

Comment:
what about putting common things here(between bfrt and tdi gen) in a function?

Reply:
Yes. will do that.

Reply:
Done